odl/util/testutils.py

```diff
@@ -94,6 +94,19 @@ def almost_equal(a, b, places=None):
 
 def all_equal(iter1, iter2):
     """Return ``True`` if all elements in ``a`` and ``b`` are equal."""
+    # Transfer cupy arrays to CPU for faster comparison
+    from odl.space.cupy_tensors import CUPY_AVAILABLE, CupyTensor, cupy
+    if CUPY_AVAILABLE:
+        if isinstance(iter1, CupyTensor):
+            iter1 = iter1.asarray()
+        elif isinstance(iter1, cupy.ndarray):
+            iter1 = cupy.asnumpy(iter1)
+
+        if isinstance(iter2, CupyTensor):
+            iter2 = iter2.asarray()
+        elif isinstance(iter2, cupy.ndarray):
+            iter2 = cupy.asnumpy(iter2)
```

Comment:
Should use `utility.asarray` here I guess.

Reply:
yes